script/deploy/facets/DeployReceiver.s.sol

```diff
@@ -36,17 +36,20 @@ contract DeployScript is DeployScriptBase {
 
         // obtain address of Stargate router in current network from config file
         string memory path = string.concat(root, "/config/stargate.json");
+        string memory stargateJson = vm.readFile(path);
 
-        address stargateRouter = _getConfigContractAddress(
-            path,
+        // we cannot use our helper function here since we sometimes deploy with address(0) where stargate is not available and this will fail
+        address stargateRouter = stargateJson.readAddress(
```

Comment:
instead of bypassing _getConfigContractAddress, we could introduce an optional allowZero parameter to let it return address(0) when needed

Reply:
that's a good idea...will do

Reply:
maybe instead of having false everywhere for allowZeroAddress we could use overloading and treat allowZeroAddress as optional. Like:

```
function _getConfigContractAddress(
    string memory path,
    string memory key
) internal returns (address contractAddress) {
    return _getConfigContractAddress(path, key, false); // where _getConfigContractAddress with three params is with allowZeroAddress
}
```